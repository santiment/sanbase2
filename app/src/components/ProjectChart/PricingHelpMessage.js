import React from 'react'
import { pure } from 'recompose'
import HelpPopup from './../../components/HelpPopup/HelpPopup'
import './PricingHelpMessage.css'

const PricingHelpMessage = (isPremium = false) => {
  if (!isPremium) {
    return ''
  }
  return (
    <div className='pricing-details'>
      <label className='pricing-details__label'>Limited data.</label>
      &nbsp;Unlock the full app. &nbsp;<HelpPopup
        trigger={<span>Learn more</span>}
      >
        <div>
          <p>
            <strong>We are free now for general usage.</strong>
          </p>
          <p>
            <strong>SANbase:</strong> 1000 SAN tokens staking
          </p>
          <ul>
            <li>Historical data more than 3 month</li>
            <li>Social: Project sentiment feed</li>
            <li>Insights voting</li>
          </ul>
          <p>
            <strong>API:</strong> 1000 SAN tokens staking
          </p>
          <ul>
            <li>Realtime data</li>
          </ul>
          <hr />
          <strong>How do I stake SAN?</strong>
          <ul>
            <li>Get Metamask and create an ETH address</li>
            <li>
              Obtain $SAN tokens from any of the exchanges listed here:
              https://coinmarketcap.com/currencies/santiment/#markets
            </li>
            <li>
              Transfer $SAN tokens from exchange to your Metamask ETH address.
              Note: Be sure to have required amount of SAN for access.
            </li>
            <li>Sign in to desired platform with Metamask for access.</li>
          </ul>
        </div>
      </HelpPopup>
    </div>
  )
}

export default pure(PricingHelpMessage)
