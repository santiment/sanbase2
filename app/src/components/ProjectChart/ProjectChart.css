.project-dp-chart {
  display: flex;
  flex-direction: column;
  position: relative;
}

.project-dp-chart .graph {
  flex: 1;
}

.project-chart-loader {
  padding-top: 10px;
}

.project-chart-body {
  position: relative;
}

.project-chart__isLoading,
.project-chart__isEmpty {
  color: black;
  font-size: 2em;
  font-weight: 300;
  position: absolute;
  display: flex;
  align-items: center;
  height: 100%;
  width: 100%;
  background: rgba(227, 227, 227, 0.4);
  border-radius: 4px;
  justify-content: center;
}

.chart-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 20px;
  min-height: 60px;
}

.chart-datetime-settings {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.time-filter {
  margin-right: 20px;
}

.selected-value {
  font-size: 1em;
  text-align: center;
}

.selected-value-datetime {
  font-size: 1.4em;
  font-weight: 300;
}

.selected-value-data {
  display: flex;
  flex-direction: column;
  font-size: 1em;
  font-weight: bold;
  margin-right: 0.5em;
  text-align: left;
}

@media (max-width: 1024px) {
  .selected-value-datetime {
    font-size: 1em;
    font-weight: 300;
  }

  .selected-value-data {
    display: flex;
    flex-direction: column;
    font-size: 0.8em;
    font-weight: bold;
    margin-right: 0;
    text-align: right;
  }
}

.chart-footer {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 10px;
}

.chart-footer-filters {
  display: flex;
  align-items: center;
}

.trademark {
  color: #e3e3e3;
}

.time-filter,
.currency-filter,
.chart-footer-filters {
  display: flex;
  user-select: none;
}

.time-filter div,
.currency-filter div,
.chart-footer-filters div {
  cursor: pointer;
  width: 40px;
  text-align: center;
}

.time-filter .activated,
.currency-filter .activated,
.chart-footer-filters .activated {
  background: gray;
  color: white;
}

.time-filter .disabled,
.currency-filter .disabled,
.chart-footer-filters .disabled {
  color: #888;
}

.chart-footer-filters div {
  width: 100px;
  margin-right: 2px;
}
