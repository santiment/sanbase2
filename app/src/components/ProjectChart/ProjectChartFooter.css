.chart-footer {
  border-top: 1px solid rgb(216, 216, 218);
}

.toggleBtn {
  display: flex;
  position: relative;
  padding: 2px 10px;
  align-items: center;
  justify-content: flex-end;
  user-select: none;
  cursor: pointer;
  text-align: center;
  border-radius: 4px;
  box-shadow: 0 4px 6px rgba(50, 50, 92, 0.11), 0 1px 3px rgba(0, 0, 0, 0.075);
}

.toggleBtn div i {
  margin-right: 0;
}

.toggleBtn .label {
  margin-right: 5px !important;
  font-size: 10px;
}

.toggleBtn .icon {
  margin-bottom: 6px;
  margin-left: 3px;
}

.toggleBtn.activated {
  background: rgba(38, 43, 51, 0.8);
  color: white;
}

.toggleBtn.disabled {
  color: #888;
}

.toggleBtn .toggleBtn--loading {
  color: transparent !important;
  text-shadow: 0 0 3px rgba(0, 0, 0, 0.3) !important;
}

.toggleBtn .toggleBtn-loader {
  position: absolute;
  top: 50%;
  left: 40%;
  color: white;
}

.marketcapLabel {
  color: rgb(52, 118, 153) !important;
  background-color: rgb(52, 118, 153) !important;
  border-color: rgb(52, 118, 153) !important;
}

.volumeLabel {
  color: rgba(38, 43, 51, 0.25) !important;
  background-color: rgba(38, 43, 51, 0.25) !important;
  border-color: rgba(38, 43, 51, 0.25) !important;
}

.githubActivityLabel {
  color: rgb(96, 76, 141) !important;
  background-color: rgb(96, 76, 141) !important;
  border-color: rgb(96, 76, 141) !important;
}

.burnRateLabel {
  color: rgb(252, 138, 23) !important;
  background-color: rgb(252, 138, 23) !important;
  border-color: rgb(252, 138, 23) !important;
}

.transactionVolumeLabel {
  color: rgb(39, 166, 153) !important;
  background-color: rgb(39, 166, 153) !important;
  border-color: rgb(39, 166, 153) !important;
}

.twitterLabel {
  color: rgb(16, 195, 245) !important;
  background-color: rgb(16, 195, 245) !important;
  border-color: rgb(16, 195, 245) !important;
}

.sentimentLabel {
  color: #e23ab4 !important;
  background-color: #e23ab4 !important;
  border-color: #e23ab4 !important;
}

.ethSpentOverTimeLabel {
  color: #c82f3f !important;
  background-color: #c82f3f !important;
  border-color: #c82f3f !important;
}

.chart-footer-filters p {
  margin: 5px !important;
}

.filter-category {
  display: flex;
  padding: 0;
  margin: 15px 0;
  border-right: 1px solid rgb(231, 231, 231);
  user-select: none;
  justify-content: space-between;
  align-items: normal;
}

.night-mode .filter-category {
  border-color: #2d3239;
  border-radius: 0;
}

.filter-category:last-child {
  border: none;
}

.filter-category-title {
  margin: 0;
  font-weight: 400;
}

.filter-category-body {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  max-width: 30%;
}

.filter-category-body .toggleBtn:last-child {
  margin-right: 0;
}

.filter-category-blockchain .filter-category-body {
  flex-direction: column;
  justify-content: end;
}

.filter-category-blockchain .filter-category-title {
  flex: 1;
  text-align: left;
}

.blockchain-settings {
  display: flex;
  flex-direction: column;
}

.blockchain-settings .checkbox {
  margin-bottom: 5px;
}

.blockchain-settings .checkbox:last-child {
  margin-bottom: 0;
}

.premium-wall-button {
  border: 2px solid gold;
  margin-top: 5px;
}

.night-mode .chart-footer-filters .toggleBtn {
  border: 1px solid #252932;
}

.night-mode .chart-footer-filters .premium-wall-button {
  border-color: #a18825;
  border-width: 2px;
}

@media (min-width: 1024px) {
  .chart-footer {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  .chart-footer-filters {
    display: flex;
  }

  .filter-category {
    display: block;
    padding: 5px 20px;
  }

  .filter-category-title {
    margin-bottom: 10px;
    flex: 1;
  }

  .filter-category-body {
    max-width: 100%;
  }

  .filter-category-blockchain {
    justify-content: end;
  }

  .filter-category-blockchain .filter-category-title {
    text-align: center;
  }

  .filter-category-blockchain .filter-category-body {
    justify-content: end;
  }

  .toggleBtn {
    justify-content: flex-start;
  }

  .filter-category-body .toggleBtn {
    margin-top: 10px;
  }

  .filter-category-body .toggleBtn:first-child {
    margin-top: 0;
  }
}
