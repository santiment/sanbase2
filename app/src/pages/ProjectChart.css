.project-dp-chart {
  display: flex;
  flex-direction: column;
  position: relative;
}

.project-dp-chart .graph {
  flex: 1;
}

.chart-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 10px;
  padding: 0 20px;
  min-height: 40px;
}

.selected-value {
  font-size: 1.2em;
  text-align: center;
}

.selected-value-datetime {
  font-size: 1.6em;
  font-weight: 300;
}

.selected-value-data {
  font-size: 1.2em;
  font-weight: bold;
  margin-right: 0.5em;
}

.time-filter,
.currency-filter {
  display: flex;
  width: 200px;
}

.time-filter div,
.currency-filter div {
  cursor: pointer;
  width: 40px;
  margin-right: 5px;
  text-align: center;
}

.time-filter .activated,
.currency-filter .activated {
  background: gray;
  color: white !important;
  width: 40px;
}

.time-filter .disabled,
.currency-filter .disabled {
  color: #888;
}

.currency-filter {
  margin-top: 10px;
}
