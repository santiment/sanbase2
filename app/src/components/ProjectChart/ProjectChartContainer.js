import React, { Component } from 'react'
import { withApollo } from 'react-apollo'
import gql from 'graphql-tag'
import moment from 'moment'
import ProjectChart from './ProjectChart'

const getHistoryGQL = gql`
  query history($ticker: String, $from: DateTime, $to: DateTime, $interval: String) {
    historyPrice(
      ticker: $ticker,
      from: $from,
      to: $to,
      interval: $interval
    ) {
      priceBtc,
      priceUsd,
      volume,
      datetime,
      marketcap
    }
}`

export const makeItervalBounds = interval => {
  switch (interval) {
    case '1d':
      return {
        from: moment().subtract(1, 'd').utc().format('YYYY-MM-DD') + 'T00:00:00Z',
        to: moment().utc().format(),
        minInterval: '5m'
      }
    case '1w':
      return {
        from: moment().subtract(1, 'weeks').utc().format(),
        to: moment().utc().format(),
        minInterval: '1h'
      }
    case '2w':
      return {
        from: moment().subtract(2, 'weeks').utc().format(),
        to: moment().utc().format(),
        minInterval: '1h'
      }
    default:
      return {
        from: moment().subtract(1, 'months').utc().format(),
        to: moment().utc().format(),
        minInterval: '1h'
      }
  }
}

const fetchPriceHistory = (client, ticker, interval = '1m') => {
  const { from, to, minInterval } = makeItervalBounds(interval)
  return new Promise((resolve, reject) => {
    client.query({
      query: getHistoryGQL,
      variables: {
        'ticker': ticker,
        'from': from,
        'to': to,
        'interval': minInterval
      }
    })
    .then(response => {
      resolve(response.data)
    })
    .catch(error => reject(error))
  })
}

class ProjectChartContainer extends Component {
  state = {
    interval: '1m',
    isLoading: true,
    isError: false,
    isEmpty: true,
    errorMessage: '',
    selected: null,
    history: []
  }

  componentDidMount () {
    const { client, ticker } = this.props
    const { interval } = this.state
    fetchPriceHistory(client, ticker, interval).then(data => {
      const historyPrice = data.historyPrice || []
      this.setState({
        isLoading: false,
        isEmpty: historyPrice.length === 0,
        history: data.historyPrice
      })
    })
    .catch(error => console.log(error))
  }

  render () {
    const { isEmpty, isLoading } = this.state.history
    return (
      <div style={{width: '100%'}}>
        {!isLoading && isEmpty
          ? 'We can\'t find any project with such ticker symbol.'
          : <ProjectChart
            setFilter={this.setFilter}
            setSelected={this.setSelected}
            {...this.state} />}
      </div>
    )
  }

  setSelected = selected => {
    this.setState({selected})
  }

  setFilter = interval => {
    if (interval === this.state.interval) { return }
    const { client, ticker } = this.props
    this.setState({
      interval,
      isLoading: true,
      selected: null
    })
    fetchPriceHistory(client, ticker, interval).then(data => {
      const historyPrice = data.historyPrice || []
      this.setState({
        isLoading: false,
        isEmpty: historyPrice.length === 0,
        history: data.historyPrice,
        selected: null
      })
    })
  }
}

export default withApollo(ProjectChartContainer)
