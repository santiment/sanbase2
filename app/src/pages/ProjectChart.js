import React from 'react'
import { graphql } from 'react-apollo'
import gql from 'graphql-tag'
import {
  compose,
  pure,
  withState,
  withHandlers
} from 'recompose'
import { Merge } from 'animate-components'
import { fadeIn, slideUp } from 'animate-keyframes'
import { Bar } from 'react-chartjs-2'
import moment from 'moment'
import { formatNumber, formatBTC } from '../utils/formatting'
import './ProjectChart.css'

const TimeFilterItem = ({disabled, filter, setFilter, value = '1d'}) => {
  let cls = filter === value ? 'activated' : ''
  if (disabled) {
    cls += ' disabled'
  }
  return (
    <div
      className={cls}
      onClick={() => !disabled && setFilter(value)}>{value}</div>
  )
}

const TimeFilter = props => (
  <div className='time-filter'>
    <TimeFilterItem value={'1d'} {...props} />
    <TimeFilterItem value={'1w'} {...props} />
    <TimeFilterItem value={'2w'} {...props} />
    <TimeFilterItem value={'1m'} {...props} />
  </div>
)

const CurrencyFilter = ({isToggledBTC, showBTC, showUSD}) => (
  <div className='currency-filter'>
    <div
      className={isToggledBTC ? 'activated' : ''}
      onClick={showBTC}>BTC</div>
    <div
      className={!isToggledBTC ? 'activated' : ''}
      onClick={showUSD}>USD</div>
  </div>
)

const getChartDataFromHistory = (history = [], isToggledBTC) => {
  return {
    labels: history ? history.map(data => new Date(data.datetime)) : [],
    datasets: [{
      label: 'price',
      type: 'line',
      strokeColor: '#7a9d83eb',
      borderColor: '#7a9d83eb',
      borderWidth: 1,
      backgroundColor: 'rgba(239, 242, 236, 0.5)',
      pointBorderWidth: 2,
      yAxisID: 'y-axis-1',
      data: history ? history.map(data => {
        if (isToggledBTC) {
          const price = parseFloat(data.priceBtc)
          return formatBTC(price)
        }
        return data.priceUsd
      }) : []
    },
    {
      label: 'volume',
      fill: false,
      type: 'bar',
      yAxisID: 'y-axis-2',
      strokeColor: 'rgba(49, 107, 174, 0.5)',
      borderColor: 'rgba(49, 107, 174, 0.5)',
      borderWidth: 1,
      backgroundColor: 'rgba(239, 242, 236, 0.5)',
      pointBorderWidth: 2,
      data: history ? history.map(data => {
        if (isToggledBTC) {
          return []
        }
        return parseFloat(data.volume)
      }) : []
    }]
  }
}

export const ProjectChart = ({
  history,
  setSelected,
  selected,
  ...props
}) => {
  if (!history || history.isLoading) {
    return (
      <h2>Loading...</h2>
    )
  }
  if (history.isError) {
    return (
      <div>
        <h2>We can't get the data from our server now... ;(</h2>
        <p>{history.errorMessage}</p>
      </div>
    )
  }
  const chartData = getChartDataFromHistory(history.data, props.isToggledBTC)
  const max = Math.max(...chartData.datasets[1].data)
  const chartOptions = {
    responsive: true,
    showTooltips: false,
    pointDot: false,
    scaleShowLabels: false,
    datasetFill: false,
    scaleFontSize: 0,
    animation: false,
    pointRadius: 0,
    tooltips: {
      callbacks: {
        title: item => '',
        label: (tooltipItem, data) => {
          return props.isToggledBTC
            ? formatBTC(tooltipItem.yLabel)
            : formatNumber(tooltipItem.yLabel, 'USD')
        }
      }
    },
    legend: {
      display: false
    },
    elements: {
      point: {
        hitRadius: 2,
        hoverRadius: 2,
        radius: 0
      }
    },
    scales: {
      yAxes: [{
        id: 'y-axis-1',
        type: 'linear',
        display: true,
        position: 'left',
        ticks: {
          display: true,
          beginAtZero: true
        },
        gridLines: {
          drawBorder: true,
          display: true
        }
      }, {
        id: 'y-axis-2',
        type: 'linear',
        display: false,
        position: 'right',
        ticks: {
          max: max * 2.2
        },
        labels: {
          show: true
        }
      }],
      xAxes: [{
        type: 'time',
        time: {
          displayFormats: {
            quarter: 'MMM YYYY'
          }
        },
        ticks: {
          autoSkipPadding: 1
        },
        gridLines: {
          drawBorder: true,
          display: true
        }
      }]
    }
  }

  return (
    <div className='project-dp-chart'>
      <div className='chart-header'>
        <TimeFilter disabled {...props} />
        <div className='selected-value'>{selected &&
          <Merge
            one={{ name: fadeIn, duration: '0.3s', timingFunction: 'ease-in' }}
            two={{ name: slideUp, duration: '0.5s', timingFunction: 'ease-out' }}
            as='div'
          >
            <span className='selected-value-datetime'>{moment(chartData.labels[selected]).format('MMMM DD, YYYY')}</span>
          </Merge>}</div>
        <div className='selected-value'>{selected &&
          <Merge
            one={{ name: fadeIn, duration: '0.3s', timingFunction: 'ease-in' }}
            two={{ name: slideUp, duration: '0.5s', timingFunction: 'ease-out' }}
            as='div'
          >
            <span className='selected-value-data'>Price: {props.isToggledBTC
              ? formatBTC(parseFloat(chartData.datasets[0].data[selected]))
              : formatNumber(chartData.datasets[0].data[selected], 'USD')}</span>
            <span className='selected-value-data'>Volume: {formatNumber(chartData.datasets[1].data[selected], 'USD')}</span>
          </Merge>}</div>
      </div>
      <Bar
        className='graph'
        data={chartData}
        options={chartOptions}
        redraw
        height={100}
        onElementsClick={elems => {
          elems[0] && setSelected(elems[0]._index)
        }}
        style={{ transition: 'opacity 0.25s ease' }}
      />
      <CurrencyFilter {...props} />
    </div>
  )
}

const getHistoryGQL = gql`
  query history($ticker: String, $from: DateTime, $to: DateTime, $interval: String) {
    historyPrice(
      ticker: $ticker,
      from: $from,
      to: $to,
      interval: $interval
    ) {
      priceBtc,
      priceUsd,
      volume,
      datetime,
      marketcap
    }
}`

const defaultFrom = moment().subtract(1, 'M').utc().format()
const defaultTo = moment().subtract(1, 'd').utc().format()

const mapDataToProps = ({historyPrice}) => {
  const isLoading = historyPrice.loading
  const isEmpty = !!historyPrice.project
  const isError = !!historyPrice.error
  const data = historyPrice.historyPrice
  const errorMessage = isError ? historyPrice.error.message : ''
  const project = historyPrice.project

  return {history: {isLoading, isEmpty, isError, project, errorMessage, data}}
}

const mapPropsToOptions = ({ticker}) => {
  return {
    variables: {
      'ticker': ticker,
      'from': defaultFrom,
      'to': defaultTo,
      'interval': '1h'
    }
  }
}

const enhance = compose(
  withState('isToggledBTC', 'currencyToggle', false),
  withHandlers({
    showBTC: ({ currencyToggle }) => e => currencyToggle(true),
    showUSD: ({ currencyToggle }) => e => currencyToggle(false)
  }),
  withState('filter', 'setFilter', '1m'),
  withState('selected', 'setSelected', null),
  graphql(getHistoryGQL, {
    name: 'historyPrice',
    props: mapDataToProps,
    options: mapPropsToOptions
  }),
  pure
)

export default enhance(ProjectChart)
