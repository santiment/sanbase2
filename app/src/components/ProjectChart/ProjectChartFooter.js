import React from 'react'
import cx from 'classnames'
import {
  Popup,
  Icon,
  Label,
  Loader,
  Checkbox
} from 'semantic-ui-react'
import './ProjectChartFooter.css'

export const ToggleBtn = ({
  loading,
  error = false,
  disabled,
  isToggled,
  toggle,
  children
}) => (
  <div className={cx({
    'toggleBtn': true,
    'activated': isToggled,
    'disabled': disabled || loading
  })}
    onClick={() => !disabled && !loading && toggle(!isToggled)}>
    {disabled && !error &&
      <Popup
        trigger={<div>{children}</div>}
        content="Looks like we don't have any data"
        position='bottom left'
      />}
    {!!error &&
      <Popup
        trigger={<div>{children}</div>}
        content='Looks like we have some problems with our server'
        position='bottom left'
    />}
    {!disabled && !error && children}
    {loading && <Loader active inline size='mini' />}
  </div>
)

const FilterCategory = ({
  children,
  name,
  className = '',
  settings
}) => (
  <div className={'filter-category ' + className}>
    <h5 className='filter-category-title'>
      {name.toUpperCase()}&nbsp;
      {settings && settings()}
    </h5>
    <div className='filter-category-body'>
      {children}
    </div>
  </div>
)

const ProjectChartFooter = (props) => (
  <div className='chart-footer'>
    <div className='chart-footer-filters'>
      <FilterCategory name='Financial'>
        <ToggleBtn
          isToggled={props.isToggledMarketCap}
          toggle={props.toggleMarketcap}>
          <Label circular className='marketcapLabel' empty />
          Marketcap
        </ToggleBtn>
        <ToggleBtn
          isToggled={props.isToggledVolume}
          toggle={props.toggleVolume}>
          <Label circular className='volumeLabel' empty />
          Volume
        </ToggleBtn>
      </FilterCategory>
      <FilterCategory name='Development'>
        <ToggleBtn
          loading={props.github.history.loading}
          disabled={props.github.history.items.length === 0}
          isToggled={props.isToggledGithubActivity &&
            props.github.history.items.length !== 0}
          toggle={props.toggleGithubActivity}>
          <Label circular className='githubActivityLabel' empty />
          Github Activity
        </ToggleBtn>
      </FilterCategory>
      <FilterCategory
        className='filter-category-blockchain'
        settings={() => (
          <Popup
            position='top center'
            size='large'
            trigger={<Icon name='cogs' />} on='click'>
            <div className='blockchain-settings'>
              <Checkbox
                label={{ children: 'all' }}
                value='all'
                checked={props.blockchainFilter === 'all'}
                onChange={() => props.setBlockchainFilter('all')}
                radio />
              <Checkbox
                label={{ children: 'only outliers' }}
                checked={props.blockchainFilter === 'only'}
                onChange={() => props.setBlockchainFilter('only')}
                value='only'
                radio />
              <Checkbox
                label={{ children: 'without outliers' }}
                checked={props.blockchainFilter === 'rest'}
                onChange={() => props.setBlockchainFilter('rest')}
                value='rest'
                radio />
            </div>
          </Popup>
        )}
        name='Blockchain'>
        <ToggleBtn
          loading={props.burnRate.loading}
          error={props.burnRate.error}
          disabled={props.burnRate.items.length === 0 || props.burnRate.error}
          isToggled={props.isToggledBurnRate &&
            props.burnRate.items.length !== 0}
          toggle={props.toggleBurnRate}>
          <Label circular className='burnRateLabel' empty />
          Burn Rate
          <Popup
            trigger={<Icon name='info circle' />}
            content='Token Burn Rate shows the amount of movement
            of tokens between addresses. One use for this metric is
            to spot large amounts of tokens moving after sitting for long periods of time'
            position='top left'
          />
        </ToggleBtn>
        <ToggleBtn
          loading={props.transactionVolume.loading}
          error={props.transactionVolume.error}
          disabled={props.transactionVolume.items.length === 0 ||
            props.transactionVolume.error}
          isToggled={props.isToggledTransactionVolume &&
            props.transactionVolume.items.length !== 0}
          toggle={props.toggleTransactionVolume}>
          <Label circular className='transactionVolumeLabel' empty />
          Transaction Volume
          <Popup
            trigger={<Icon name='info circle' />}
            content='Total amount of tokens that were transacted on the blockchain'
            position='top left'
          />
        </ToggleBtn>
      </FilterCategory>
      <FilterCategory name='Social'>
        <ToggleBtn
          loading={props.twitter.history.loading}
          disabled={props.twitter.history.items.length === 0}
          isToggled={props.isToggledTwitter &&
            props.twitter.history.items.length !== 0}
          toggle={props.toggleTwitter}>
          <Label circular className='twitterLabel' empty />
          Twitter
        </ToggleBtn>
      </FilterCategory>
    </div>
  </div>
)

export default ProjectChartFooter
