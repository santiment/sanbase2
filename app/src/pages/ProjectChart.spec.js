/* eslint-env jest */
import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ProjectChart } from './ProjectChart'

const historyData = [{
  priceBtc: 123,
  datetime: '2017-12-27T21:45Z'
}, {
  priceBtc: 123,
  datetime: '2017-12-27T21:45Z'
}]

describe('ProjectChart component', () => {
  it('(smoke) it should render correctly', () => {
    const chart = shallow(<ProjectChart
      history={{
        isLoading: false,
        isError: false,
        data: historyData
      }} />)
    expect(toJson(chart)).toMatchSnapshot()
  })

  describe('Loading State', () => {
    it('it should render correctly with loading prop', () => {
      const chart = mount(<ProjectChart
        history={{
          isLoading: true,
          isError: false,
          data: historyData
        }} />)
      expect(toJson(chart)).toMatchSnapshot()
    })

    it('it should render correctly withour history prop', () => {
      const chart = mount(<ProjectChart />)
      expect(toJson(chart)).toMatchSnapshot()
    })
  })

  describe('Error State', () => {
    it('it should render correctly with error message', () => {
      const chart = mount(<ProjectChart
        history={{
          isLoading: false,
          isError: true,
          errorMessage: '400 error',
          data: historyData
        }} />)
      expect(toJson(chart)).toMatchSnapshot()
    })
  })
})
