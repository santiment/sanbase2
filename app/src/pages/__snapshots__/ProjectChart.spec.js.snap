// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectChart component (smoke) it should render correctly 1`] = `
<div
  className="project-dp-chart"
>
  <div
    className="chart-header"
  >
    <TimeFilter
      disabled={true}
    />
    <div
      className="selected-value"
    />
  </div>
  <Line
    className="graph"
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "rgba(239, 242, 236, 0.5)",
            "borderColor": "#7a9d83eb",
            "borderWidth": 1,
            "data": Array [
              undefined,
              undefined,
            ],
            "pointBorderWidth": 0,
            "strokeColor": "#7a9d83eb",
          },
        ],
        "labels": Array [
          2017-12-27T21:45:00.000Z,
          2017-12-27T21:45:00.000Z,
        ],
      }
    }
    height={100}
    onElementsClick={[Function]}
    options={
      Object {
        "animation": false,
        "datasetFill": false,
        "elements": Object {
          "point": Object {
            "hitRadius": 2,
            "hoverRadius": 2,
            "radius": 0,
          },
        },
        "legend": Object {
          "display": false,
        },
        "pointDot": false,
        "pointRadius": 0,
        "responsive": true,
        "scaleFontSize": 0,
        "scaleShowLabels": false,
        "scales": Object {
          "xAxes": Array [
            Object {
              "gridLines": Object {
                "display": true,
                "drawBorder": true,
              },
              "ticks": Object {
                "autoSkipPadding": 1,
              },
              "time": Object {
                "displayFormats": Object {
                  "quarter": "MMM YYYY",
                },
              },
              "type": "time",
            },
          ],
          "yAxes": Array [
            Object {
              "gridLines": Object {
                "display": true,
                "drawBorder": true,
              },
              "ticks": Object {
                "display": true,
              },
            },
          ],
        },
        "showTooltips": false,
        "tooltips": Object {
          "callbacks": Object {
            "label": [Function],
            "title": [Function],
          },
        },
      }
    }
    redraw={true}
    style={
      Object {
        "transition": "opacity 0.25s ease",
      }
    }
  />
  <CurrencyFilter />
</div>
`;

exports[`ProjectChart component Error State it should render correctly with error message 1`] = `
<ProjectChart
  history={
    Object {
      "data": Array [
        Object {
          "datetime": "2017-12-27T21:45Z",
          "priceBtc": 123,
        },
        Object {
          "datetime": "2017-12-27T21:45Z",
          "priceBtc": 123,
        },
      ],
      "errorMessage": "400 error",
      "isError": true,
      "isLoading": false,
    }
  }
>
  <div>
    <h2>
      We can't get the data from our server now... ;(
    </h2>
    <p>
      400 error
    </p>
  </div>
</ProjectChart>
`;

exports[`ProjectChart component Loading State it should render correctly with loading prop 1`] = `
<ProjectChart
  history={
    Object {
      "data": Array [
        Object {
          "datetime": "2017-12-27T21:45Z",
          "priceBtc": 123,
        },
        Object {
          "datetime": "2017-12-27T21:45Z",
          "priceBtc": 123,
        },
      ],
      "isError": false,
      "isLoading": true,
    }
  }
>
  <h2>
    Loading...
  </h2>
</ProjectChart>
`;

exports[`ProjectChart component Loading State it should render correctly withour history prop 1`] = `
<ProjectChart>
  <h2>
    Loading...
  </h2>
</ProjectChart>
`;
